dev/TEAL_opcodes.md

```diff
@@ -522,20 +531,64 @@ for notes on transaction fields available, see `txn`. If top of stack is _i_, `g
 - push Ith value of the array field F from the Ath transaction in the current group
 - LogicSigVersion >= 3
 
+## gload t i
+
+- Opcode: 0x3a {uint8 transaction group index} {uint8 position in scratch space to load from}
+- Pops: _None_
+- Pushes: any
+- push Ith scratch space index of the Tth transaction in the current group
+- LogicSigVersion >= 4
+- Mode: Application
+
+The `gload` opcode can only access scratch spaces of previous app calls contained in the current group.
+
+## gloads i
+
+- Opcode: 0x3b {uint8 position in scratch space to load from}
+- Pops: *... stack*, uint64
+- Pushes: any
+- push Ith scratch space index of the Ath transaction in the current group
+- LogicSigVersion >= 4
+- Mode: Application
+
+The `gloads` opcode can only access scratch spaces of previous app calls contained in the current group.
+
+## gaid t
+
+- Opcode: 0x3c
+- Pops: _None_
+- Pushes: uint64
+- push the ID of the asset or application created in the Tth transaction of the current group
+- LogicSigVersion >= 4
+- Mode: Application
+
+The `gaid` opcode can only access the ID of assets or applications created by previous txns in the current group.
+
+## gaids
+
+- Opcode: 0x3d
+- Pops: *... stack*, uint64
+- Pushes: uint64
+- push the ID of the asset or application created in the Ath transaction of the current group
+- LogicSigVersion >= 4
+- Mode: Application
+
+The `gaids` opcode can only access the ID of assets or applications created by previous txns in the current group.
+
 ## bnz target
 
-- Opcode: 0x40 {0..0x7fff forward branch offset, big endian}
+- Opcode: 0x40 {int16 branch offset, big endian. (negative offsets are illegal before v4)}
```

Comment:
Maybe rephrase the parenthesized sentence: "(negative offsets are only allowed starting v4, they are illegal in prior versions)"

Reply:
I removed that explanation there, as that should really be a concise explanation of the encoding of the immediate.  The `b` instruction explains the versioning rules, and I've just updated it to be a bit clearer (to be pushed very soon)